substrate/frame/nomination-pools/src/migration.rs

```diff
@@ -109,6 +118,65 @@ pub mod unversioned {
 	}
 }
 
+mod v9 {
+	use super::*;
+
+	#[derive(Encode, Decode, MaxEncodedLen, Clone, Copy, Debug, PartialEq, Eq, TypeInfo)]
+	pub enum OldClaimPermission {
+		Permissioned,
+		PermissionlessCompound,
+		PermissionlessWithdraw,
+		PermissionlessAll,
+	}
+
+	impl OldClaimPermission {
+		fn migrate_to_v9(self) -> Option<ClaimPermission> {
+			match self {
+				OldClaimPermission::Permissioned => Some(ClaimPermission::Permissioned),
+				OldClaimPermission::PermissionlessCompound =>
+					Some(ClaimPermission::PermissionlessCompound),
+				OldClaimPermission::PermissionlessWithdraw =>
+					Some(ClaimPermission::PermissionlessWithdraw),
+				OldClaimPermission::PermissionlessAll =>
+					Some(ClaimPermission::PermissionlessWithdraw),
+			}
+		}
+	}
+
+	pub struct VersionUncheckedMigrateV8ToV9<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> OnRuntimeUpgrade for VersionUncheckedMigrateV8ToV9<T> {
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, TryRuntimeError> {
+			let claim_permission_count = ClaimPermissions::<T>::iter_keys().count();
+			Ok((claim_permission_count as u64).encode())
+		}
+
+		fn on_runtime_upgrade() -> Weight {
+			let mut translates = 0u64;
+			ClaimPermissions::<T>::translate::<OldClaimPermission, _>(|_key, current_val| {
```

Comment:
Since its only a small subset of accounts that has `PermissionlessAll` variant, technically you could avoid rewriting the other variants (since they are same) and translate only the `PermissionlessAll`. Not sure the best way to do it and since its one time its okay to do like this as well. 

P.S.: Most probably the overlay cache optimises this and does not rewrite if its the same value. But we are pedantic we can avoid hitting the overlay cache as well.